### PR TITLE
[16.0][IMP] loyalty_initial_date_validity: add test and change development status

### PR DESCRIPTION
**Pull request by pilarvargas-tecnativa**
cc @Tecnativa TT45685

@pedrobaeza @chienandalu please review

### CODE REVIEW

File: loyalty_initial_date_validity/tests/test_loyalty_initial_date_validity.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+                            "discount_mode": "percent",
+                            "discount_applicability": "order",
+                        },
+                    )
+                ],
+            }
+        )
+
+    def test_01_check_date_from_date_to(self):
+        # import wdb; wdb.set_trace()
```
Comment (chienandalu) [inline loyalty_initial_date_validity/tests/test_loyalty_initial_date_validity.py]: :point_up: 
```suggestion
```

File: loyalty_initial_date_validity/tests/test_loyalty_initial_date_validity.py [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+        self.assertFalse(self.promotion.date_to)
+        # The end date cannot be earlier than the start date.
+        with self.assertRaises(UserError):
+            self.promotion.date_to = date.today() - timedelta(days=7)
+        self.promotion.date_to = date.today() + timedelta(days=7)
```
Comment (chienandalu) [inline loyalty_initial_date_validity/tests/test_loyalty_initial_date_validity.py]: Relative dates in tests are hard to read. It's better to use `freezegun` and fixed dates.

**Review by chienandalu — CHANGES_REQUESTED** [review]

File: loyalty_initial_date_validity/tests/test_loyalty_initial_date_validity.py (hunk 2 above)
Comment (pilarvargas-tecnativa) [reply loyalty_initial_date_validity/tests/test_loyalty_initial_date_validity.py]: done, thanks!

**Review by pedrobaeza — APPROVED** [review]

File: loyalty_initial_date_validity/tests/test_loyalty_initial_date_validity.py [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+from odoo.addons.base.tests.common import DISABLED_MAIL_CONTEXT
+
+
+@tagged("-at_install", "post_install")
+class LoyaltyInitialDateValidity(TransactionCase):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.env = cls.env(context=dict(cls.env.context, **DISABLED_MAIL_CONTEXT))
```
Comment (victoralmau) [inline loyalty_initial_date_validity/tests/test_loyalty_initial_date_validity.py]:
```suggestion

from odoo.addons.base.tests.common import BaseCommon


@tagged("-at_install", "post_install")
class LoyaltyInitialDateValidity(BaseCommon):
    @classmethod
    def setUpClass(cls):
        super().setUpClass()
```

**Review by victoralmau — APPROVED** [review]

File: loyalty_initial_date_validity/tests/test_loyalty_initial_date_validity.py (hunk 3 above)
Comment (pilarvargas-tecnativa) [reply loyalty_initial_date_validity/tests/test_loyalty_initial_date_validity.py]: thanks!

**Review by chienandalu — APPROVED** [review]
:ok_hand:

**Pull request merged**